config/samples/metal3.io_v1alpha1_hardwareclassification.yaml

```diff
@@ -8,18 +8,25 @@ metadata:
 spec:
   hardwareCharacteristics:
       cpu:
-         minimumCount: 48
+         minimumCount: 2
          maximumCount: 72
-         minimumSpeedMHz: 2600
+         minimumSpeedMHz: 1000
          maximumSpeedMHz: 3600
       disk:
          minimumCount: 1
          maximumCount: 8
-         minimumIndividualSizeGB: 200
+         minimumIndividualSizeGB: 20
          maximumIndividualSizeGB: 3000
       ram:
-         minimumSizeGB: 6
+         minimumSizeGB: 2
          maximumSizeGB: 180
       nic:
          minimumCount: 1
          maximumCount: 7
+      firmware:
+         bios:
+            vendor: "Dell Inc."
+            version: "1.5.6"
```

Comment:
I think instead of asking for specific version, Can we define minor and major version. That way, I think clasiffication make more sense.
@dhellmann  what do you think ?

Because Hardware firmware are getting updated more often, I think we define like below -

      firmware:
         bios:
            vendor: "Dell Inc."
            minorVersion: "1.5.5"
            majorVersion: "1.5.8

What about this ?

Reply:
Comparing versions against a range would be useful, but can be pretty complicated. Do all firmware vendor versions have only integers separated by `.`?

What happens if a user specifies `minimumVersion: 1.5` but the actual version is `1.5.1`? Or `v1.5`?

What if they say `maximumVersion: 1.5` and the actual version is `1.5.1`?

I think we could define some reasonable behavior for those cases, if having a range of values is sufficiently important. It would be useful to see if there are any existing tools that do similar comparisons to understand how they work.

Reply:
Versioning has lots of combinations. I think current approach looks good. 
We can keep this current approach for now, this is solving most of the stuff. If there is specific requirement comes in the future, we can add more stuff to this.